src/libstd/sync/mpsc/mod.rs

```diff
@@ -591,8 +591,10 @@ impl<T: Send> Sender<T> {
                                 // This send cannot panic because the thread is
                                 // asleep (we're looking at it), so the receiver
                                 // can't go away.
-                                (*a.get()).send(t).ok().unwrap();
-                        token.signal();
+                                if let Err(_) = (*a.get()).send(t) {
```

Comment:
this is really ugly, but it represents the comment more.
maybe I should just leave the `ok().unwrap()`?